src/test/java/org/jenkinsci/plugins/pipeline/modeldefinition/PostStageTest.java

```diff
@@ -0,0 +1,236 @@
+
+    @Test
+    public void withAllLocalUnstable() throws Exception {
+        env().put("MAKE_RESULT", Result.UNSTABLE.toString()).set();
+        expect(Result.UNSTABLE, "localAll").logContains(ALL_LOCAL_ALWAYS)
+                .logContains("Setting build result UNSTABLE", "I AM UNSTABLE", "I HAVE CHANGED")
+                .logNotContains("I WAS ABORTED", "I FAILED", "MOST DEFINITELY FINISHED").go();
+
+    }
+
+    @Test
+    public void withAllLocalFailure() throws Exception {
+        env().put("MAKE_RESULT", Result.FAILURE.toString()).set();
+        expect(Result.FAILURE, "localAll").logContains(ALL_LOCAL_ALWAYS)
+                .logContains("Setting build result FAILURE", "I FAILED", "I HAVE CHANGED")
+                .logNotContains("I WAS ABORTED", "I AM UNSTABLE", "MOST DEFINITELY FINISHED").go();
+
+    }
+
+    @Test
+    public void withAllLocalAborted() throws Exception {
+        env().put("MAKE_RESULT", Result.ABORTED.toString()).set();
+        expect(Result.ABORTED, "localAll").logContains(ALL_LOCAL_ALWAYS)
+                .logContains("Setting build result ABORTED", "I WAS ABORTED", "I HAVE CHANGED")
+                .logNotContains("I FAILED", "I AM UNSTABLE", "MOST DEFINITELY FINISHED").go();
+
+    }
+
+    @Test
+    public void withAllLocalSuccess() throws Exception {
+        env().set();
+        expect(Result.SUCCESS, "localAll").logContains(ALL_LOCAL_ALWAYS)
+                .logContains("All is well", "MOST DEFINITELY FINISHED", "I HAVE CHANGED")
+                .logNotContains("I WAS ABORTED", "I FAILED", "I AM UNSTABLE").go();
+
+    }
+
+    @Test
+    public void withAllLocalChanged() throws Exception {
+        env().set();
+        ExpectationsBuilder expect = expect(Result.SUCCESS, "localAll").logContains(ALL_LOCAL_ALWAYS);
+        expect.logContains("All is well", "MOST DEFINITELY FINISHED", "I HAVE CHANGED")
+                .logNotContains("I WAS ABORTED", "I FAILED", "I AM UNSTABLE").go();
+        expect.resetForNewRun(Result.SUCCESS)
+                .logContains("All is well", "MOST DEFINITELY FINISHED")
+                .logNotContains("I WAS ABORTED", "I FAILED", "I AM UNSTABLE", "I HAVE CHANGED").go();
+
+        env().put("MAKE_RESULT", Result.UNSTABLE.toString()).set();
+        expect.resetForNewRun(Result.UNSTABLE).logContains(ALL_LOCAL_ALWAYS)
+                .logContains("Setting build result UNSTABLE", "I AM UNSTABLE", "I HAVE CHANGED")
+                .logNotContains("I WAS ABORTED", "I FAILED", "MOST DEFINITELY FINISHED").go();
+
+    }
+
+    @Test
+    public void failingNotifications() throws Exception {
+        expect(Result.FAILURE, "failingNotifications")
+                .logContains("hello", "Post stage", "farewell", "Error in stage post:", "I AM HERE STILL")
+                .logNotContains("world").go();
+    }
+
+
+    //Helper methods and builders, could possibly be moved to something more central for reuse
```

Comment:
Ok, I change my mind, :bug: until this awesome stuff is moved to `AbstractModelDefTest`. =)